### PR TITLE
NSM Test suites

### PR DESCRIPTION
**Pull request by denis-tingajkin**
## Description
**Be free to comment.
Not ready for merge yet.
Work in progress.**

## Motivation and Context
Add nsm test suites on ci to speedup test time.

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .cloudtest.yaml [hunk 1]
```diff
@@ -13,183 +13,197 @@ import:
 reporting:
   junit-report: "results/junit.xml"
 executions:
-  - name: "Single cluster tests"
+  - name: "Single cluster tests uses shared nsm infrastructure"
     env:
-      - STORE_POD_LOGS_IN_FILES=true
-      - STORE_POD_LOGS_DIR=/home/circleci/project/.tests/cloud_test/$(cluster-name)
+      - PROCESS_ARTIFACTS_IN_TO_ARCHIVE=true
```
Comment (alex-yust) [inline .cloudtest.yaml]: PROCESS_ARTIFACTS_INTO_ARCHIVE? 
ARCHIVE_ARTIFACTS?

Comment (denis-tingajkin) [reply .cloudtest.yaml]: sure.
PR has mark work in progress

File: test/cloudtest/pkg/commands/commands.go [hunk 2]
```diff
@@ -171,11 +173,35 @@ func CloudTestRun(cmd *cloudTestCmd) {
...
+			dir := imp[:len(imp)-1]
+			files, err := ioutil.ReadDir(dir)
+			if err != nil {
+				return err
+			}
+			var fileNames []string
+			for _, f := range files {
+				fileNames = append(fileNames, filepath.Join(dir, f.Name()))
+			}
+			if err := importFiles(testConfig, fileNames...); err != nil {
```
Comment (alex-yust) [inline test/cloudtest/pkg/commands/commands.go]: How about making testConfig a receiver?
testConfig.importFiles(files...) looks nicer, IMO

**Comment by denis-tingajkin** [comment]
this PR was split into 4 parts:
1) https://github.com/networkservicemesh/cloudtest/pull/3
2) https://github.com/networkservicemesh/networkservicemesh/pull/2057
3) https://github.com/networkservicemesh/networkservicemesh/pull/2042
4) https://github.com/networkservicemesh/networkservicemesh/pull/2061

**Pull request closed without merging**